src/mockServiceWorker.js

```diff
@@ -309,6 +313,22 @@ function sendToClient(client, message) {
         return reject(event.data.error)
       }
 
+      switch (event.data?.type) {
```

Comment:
You can't use optional chaining, as well as many other modern ES features, in the worker script. This is also why the smoke test fails (older Chromium browsers don't support this syntax). I didn't know the newer browsers support this, it's neat.

Reply:
looks like object spread was already used in `{...clientMessage.payload}`, so I validated the script in https://eslint.org/demo using version ES2018